File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        total_a = sum(apple)
        total_c = sum(capacity)
        count = 0
        if total_a >= total_c:
            return len(capacity)
        else:
            while total_a > 0:
                capacity.sort()
                total_a -= capacity[-1]
                capacity.pop()
                count += 1
            return count

File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.js
/**
 * @param {number[]} apple
 * @param {number[]} capacity
 * @return {number}
 */
var minimumBoxes = function(apple, capacity) {
    const totalApple = apple.reduce((accu, curr) => accu + curr, 0);
    console.log(totalApple)
    const capacitySorted = capacity.sort((a, b) => b - a);
    let currentCapacity = 0;
    for (let i = 0; i < capacitySorted.length; i++) {
        currentCapacity += capacitySorted[i];
        if(currentCapacity >= totalApple) {
            return i + 1;
        }
    }
};